### PR TITLE
freebsd proposal to move sockcred2 usage in the root mod while ignori…

### PR DESCRIPTION
**Pull request by devnexen**
…ng it for FreeBSD 12.

### CODE REVIEW

**Comment by Amanieu** [comment]
cc @asomers @GuillaumeGomez

File: src/unix/bsd/freebsdlike/freebsd/mod.rs [hunk 1]
```diff
@@ -3701,6 +3712,15 @@ f! {
         let (idx, offset) = (cpu / bitset_bits, cpu % bitset_bits);
         0 != cpuset.__bits[idx] & (1 << offset)
     }
+
+    pub fn SOCKCRED2SIZE(ngrps: usize) -> usize {
```
Comment (GuillaumeGomez) [inline src/unix/bsd/freebsdlike/freebsd/mod.rs:3716]: Won't it be an issue on freebsd 11 to link to a non-existing function?

Comment (GuillaumeGomez) [reply src/unix/bsd/freebsdlike/freebsd/mod.rs:3716]: Ah no, nevermind me...

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me since the struct has the same layout between 13 and 14 and the function is implemented, not linked.

**Review by asomers — APPROVED** [review]
LGTM.  This is the best way to do with new symbols in the OS, rather than changed symbols.

**Pull request merged**